code/modules/goonchat/browserOutput.dm

```diff
@@ -180,17 +197,26 @@ GLOBAL_DATUM_INIT(iconCache, /savefile, new("tmp/iconCache.sav")) //Cache of ico
 		if (connData && islist(connData) && connData.len > 0 && connData["connData"])
 			connectionHistory = connData["connData"] //lol fuck
 			var/list/found = new()
+
+			if(connectionHistory.len > MAX_COOKIE_LENGTH)
+				message_admins("[key_name(src.owner)] was kicked for an invalid ban cookie)")
+				qdel(owner)
+				return
+
 			for(var/i in connectionHistory.len to 1 step -1)
+				if(QDELETED(owner))
```

Comment:
```suggestion
				if(CHECK_TICK && QDELETED(owner))
```

Personally I always prefer the top of the loop but that's personal preference, the neat thing here though is you avoid checking qdeleted if you don't sleep from checktick

Reply:
as soon as I sleep, I need to check every single entry in the list?, this would only check the first item post sleep

Reply:
ah that's true